src/components/describe/describeWebview.ts

```diff
@@ -1,23 +1,51 @@
 import * as vscode from 'vscode';
+import { Kubectl } from '../../kubectl';
+import { ShellResult } from '../../shell';
 import { WebPanel } from '../webpanel/webpanel';
 
 export class DescribePanel extends WebPanel {
     public static readonly viewType = 'vscodeKubernetesDescribe';
     public static currentPanels = new Map<string, DescribePanel>();
 
-    public static createOrShow(content: string, resource: string) {
+    public static createOrShow(content: string, resource: string, cmd: string, kubectl: Kubectl) {
         const fn = (panel: vscode.WebviewPanel, content: string, resource: string): DescribePanel => {
-            return new DescribePanel(panel, content, resource);
+            return new DescribePanel(panel, content, resource, cmd, kubectl);
         };
         WebPanel.createOrShowInternal<DescribePanel>(content, resource, DescribePanel.viewType, "Kubernetes Describe", DescribePanel.currentPanels, fn);
     }
 
     private constructor(
         panel: vscode.WebviewPanel,
         content: string,
-        resource: string
+        resource: string,
+        cmd: string,
+        kubectl: Kubectl
     ) {
         super(panel, content, resource, DescribePanel.currentPanels);
+
+        this.panel.webview.onDidReceiveMessage(
+            (message) => {
+                switch (message.command) {
+                    case 'refresh':
+                        kubectl.invokeAsync(cmd).then((result: ShellResult | void) => {
```

Comment:
nitpick:
```suggestion
                        kubectl.invokeAsync(cmd).then((result: ShellResult | undefined) => {
```
(you can probably even leave out the type annotation - I think TypeScript should infer it)